TLS Handshake error with vault agent injector

I am running the vault agent injector with auto tls enabled and configured an external vault server that is running on my host.

```
helm install vault hashicorp/vault \
    --set "injector.externalVaultAddr=http://${HOST_PRIVATE_IP}:8200"
```
Everything was working fine, suddenly after 24 hours, I am getting this bad certificate issue.

I have even tried using  `vault.hashicorp.com/tls-skip-verify` annotation but the result is the same.
These are the agent injector logs.

```
kubectl logs -f vault-agent-injector-688d969fd6-fnxg5 -n vault
2021-08-02T13:21:51.952Z [INFO]  handler: Starting handler..
2021-08-02T13:21:51.961Z [INFO]  handler.auto-tls: Generated CA
Listening on ":8080"...
2021-08-02T13:21:51.968Z [INFO]  handler.certwatcher: Updated certificate bundle received. Updating certs...
2021-08-02T14:24:33.851Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T14:29:55.076Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T14:39:55.056Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T14:40:23.280Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T14:43:21.314Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T14:43:46.477Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T14:57:04.122Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T14:57:27.107Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T15:01:42.361Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T15:02:09.823Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T15:11:48.065Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T15:21:37.688Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T15:35:41.022Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T15:43:42.305Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T15:44:04.952Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T15:45:44.332Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T15:50:16.360Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T15:51:39.813Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T15:55:00.744Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T15:55:18.417Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T16:03:29.690Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T16:04:04.751Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T16:04:25.749Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T16:08:38.321Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T16:12:49.566Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T16:19:58.982Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T16:20:17.434Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-02T16:21:46.631Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-03T07:49:19.399Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-03T07:52:39.347Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-03T10:49:37.464Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-03T10:50:45.909Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-03T11:04:06.630Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-03T11:05:08.257Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-03T11:11:42.147Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-03T11:12:00.599Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-03T11:18:34.290Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-03T11:20:16.163Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-03T11:23:51.858Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-03T13:01:44.956Z [INFO]  handler: Request received: Method=POST URL=/mutate?timeout=10s
2021-08-03T13:36:24.566Z [ERROR] handler: http: TLS handshake error from 172.17.0.1:17971: remote error: tls: bad certificate
2021-08-03T13:51:32.001Z [ERROR] handler: http: TLS handshake error from 172.17.0.1:20802: remote error: tls: bad certificate
2021-08-03T13:56:14.122Z [ERROR] handler: http: TLS handshake error from 172.17.0.1:36180: remote error: tls: bad certificate
2021-08-03T13:57:10.726Z [ERROR] handler: http: TLS handshake error from 172.17.0.1:4425: remote error: tls: bad certificate
2021-08-03T14:01:01.632Z [ERROR] handler: http: TLS handshake error from 172.17.0.1:10077: remote error: tls: bad certificate
2021-08-03T14:01:26.954Z [ERROR] handler: http: TLS handshake error from 172.17.0.1:50072: remote error: tls: bad certificate
2021-08-03T14:01:54.899Z [ERROR] handler: http: TLS handshake error from 172.17.0.1:17536: remote error: tls: bad certificate
2021-08-03T14:12:29.850Z [ERROR] handler: http: TLS handshake error from 172.17.0.1:12749: remote error: tls: bad certificate
2021-08-03T14:12:52.509Z [ERROR] handler: http: TLS handshake error from 172.17.0.1:1626: remote error: tls: bad certificate
```

--- a/subcommand/injector/command.go
+++ b/subcommand/injector/command.go
@@ -355,7 +355,11 @@ func (c *Command) certWatcher(ctx context.Context, ch <-chan cert.Bundle, client
 
 		// clear the timer
 		if !timer.Stop() {
-			<-timer.C
+			// non-blocking drain
+			select {
+			case <-timer.C:
+			default:
+			}
 		}
 
 		err := c.updateCertificate(ctx, clientset, bundle, webhooksCache, leaderElector, log)

